src/recordlinker/linkage/matchers.py

```diff
@@ -447,18 +447,18 @@ def single_feature_match_exact(
     :param record: The incoming record to evaluate.
     :param patient: The patient record to compare against.
     :param key: The name of the column being evaluated (e.g. "city").
-    :return: A boolean indicating whether the features are an exact match.
+    :return: A float indicating whether the features are an exact match.
     """
     for x in patient.record.field_iter(key):
         for y in record.field_iter(key):
             if x == y:
-                return True
-    return False
+                return 1
+    return 0
 
 
-def single_feature_match_fuzzy_string(
-    record: models.PIIRecord, patient: models.Patient, key: models.FEATURE, **kwargs: dict
-) -> bool:
+def single_feature_match_fuzzy(
+    record: models.PIIRecord, patient: models.Patient, key: models.Feature, **kwargs: dict
+) -> float:
```

Comment:
Curious why was this changed to a float instead of staying a `bool`

Reply:
As you can see from line 22, the signature of these functions changed to return a float rather than a bool.
```
FEATURE_COMPARE_FUNC = typing.Callable[[models.PIIRecord, models.Patient, models.Feature], float]
```

As I dug in deeper to the existing feature matching functions, I noticed that some of them (specifically the log_odds versions) were returning floats.  It just so happens, that the eval functions were also kind of expecting a list of numerics to determine if its a match or not.  They have been previously taking advantage of that fact that you can sum booleans and get a numeric value.  For example, `sum([True, False, True, False, False]) == 2` in python.